tests/FakeItEasy.Specs/CallMatchingSpecs.cs

```diff
@@ -351,6 +356,24 @@ public interface IHaveAnOutParameter
         }
 
         [Scenario]
+        public static void ThatArgumentConstraintForValueTypeWithNullArgument(
+            IHaveANullableParameter subject,
+            int result)
+        {
+            "Given a fake with a method that accepts a nullable value type parameter"
+                .x(() => subject = A.Fake<IHaveANullableParameter>());
+
+            "And a call configured for a non-nullable argument of that type"
+                .x(() => A.CallTo(() => subject.Bar(A<int>._)).Returns(42));
+
+            "When I make a call to this method with a null argument"
+                .x(() => result = subject.Bar(null));
+
+            "Then it doesn't match the configured call and returns the default value"

```

Comment:
If I were going to be super picky, I'd note that we could probably do without the compound condition here. There was no mention of return values in the "And" step above. I'd consider just dropping everything after the "call" in this step description.
I wouldn't hold things up for this, though.